framework/src/mfem/interfaces/MFEMBoundaryRestrictable.C

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef MFEM_ENABLED
+
+#include "MFEMBoundaryRestrictable.h"
+
+InputParameters
+MFEMBoundaryRestrictable::validParams()
+{
+  // Create InputParameters object that will be appended to the parameters for the inheriting object
+  InputParameters params = emptyInputParameters();
+  params.addParam<std::vector<BoundaryName>>(
+      "boundary",
+      {"-1"},
+      "The list of boundaries (ids or names) from the mesh where this object applies. "
+      "Defaults to all boundaries.");
+  return params;
+}
+
+MFEMBoundaryRestrictable::MFEMBoundaryRestrictable(const InputParameters & parameters,
+                                                   const mfem::ParMesh & mfem_mesh)
+  : _mfem_mesh(mfem_mesh),
+    _boundary_names(parameters.get<std::vector<BoundaryName>>("boundary")),
+    _boundary_attributes(_boundary_names.size())
+{
+  _boundary_attributes = boundariesToAttributes(_boundary_names);
+  mfem::common::AttrToMarker(
+      _mfem_mesh.bdr_attributes.Max(), _boundary_attributes, _boundary_markers);
```

Comment:
This should have the same guard to avoid failing an MFEM assertion, e.g. `if (_boundary_attributes.Size())` or `if (_boundary_names.size())`